main.py

```diff
@@ -368,20 +369,21 @@ def main(cl_arguments):
         splits_to_write = evaluate.parse_write_preds_arg(args.write_preds)
         if args.transfer_paradigm == "finetune":
             for task in target_tasks:
-                if task.name == 'mnli-diagnostic': # we'll load mnli-diagnostic during mnli
-                    continue
+                # if a task borrow classifier from another task, we load parameter from that task instead
+                if task.name != \
+                    model._get_task_params(task.name).get('use_classifier', task.name):
+                    name_to_load = model._get_task_params(task.name).get('use_classifier', task.name)
+                else:
+                    name_to_load = task.name
 
-                finetune_path = os.path.join(args.run_dir, "model_state_%s_best.th" % task.name)
+                finetune_path = os.path.join(args.run_dir, "model_state_%s_best.th" % name_to_load)
                 if os.path.exists(finetune_path):
                     ckpt_path = finetune_path
                 else:
                     ckpt_path = get_best_checkpoint_path(args.run_dir)
                 load_model_state(model, ckpt_path, args.cuda, skip_task_models=[], strict=strict)
 
-                tasks = [task]
-                if task.name == 'mnli':
-                    tasks += [t for t in target_tasks if t.name == 'mnli-diagnostic']
```

Comment:
Make sure to double check that mnli-diagnostic still works the same way.

Reply:
I can do it on Thursday or Friday.
Some part of the code in mnli-diagnostic is unnecessarily unique and long, since we need to test it either way, I wonder can we take this chance to refactor mnli-diagnostic?

Reply:
Sure, but you'd have to be careful to test it thoroughly.